package/base-files/files/lib/preinit/90_user_config_menu

```diff
@@ -0,0 +1,129 @@
+#!/bin/sh
+# SPDX-License-Identifier: GPL-2.0
+# (c) 2019, Manuel Giganto <mgigantoregistros@gmail.com>
+
+config_wait_for_key () {
+	local options_limit=$1
+	local timeout=$2
+	local buttons_file=$3
+	local keypressed
+	local buttonspressed=0
+	local wait
+
+	wait="$(mktemp)" || wait=/tmp/.wait && touch $wait
+	local undo="lock -u $wait ; rm -f $wait"
+	trap "$undo" INT
+	trap "$undo" USR1
+
+	[ -n "$timeout" ] && [ $timeout -ge 1 ] || timeout=1
+	local gettime="date +%s"
+	local now=$($gettime)
+	local deadline=$(( $now + $timeout ))
+
+	lock $wait
+	{
+		sleep $timeout
+		lock -u $wait
+	} &
+
+	{
+		while [ "$now" -lt "$deadline" ] ; do
+			wait_time=$(( $deadline - $now ))
+			user_read=""
+			read -t "$wait_time" user_read
+			if [ "$user_read" -ge 0 -a "$user_read" -le "$options_limit" ] ; then
+				keypressed=$user_read
+				break
+			fi &>/dev/null
+			now=$($gettime)
+			[ "$now" -lt "$deadline" ] && echo -n "Option not valid. Choose 0-$options_limit: "
+		done
+		lock -u $wait
+	}
+
+	lock -w $wait
+	rm -f $wait
+
+	if [ -f "$buttons_file" ] ; then
+		set -- $(wc -l $buttons_file)
+		[ "$1" -le "$options_limit" ] && buttonspressed=$1
+		rm "$buttons_file"
+	fi
+
+	[ -n "$keypressed" ] && return $keypressed || return $buttonspressed
+}
+
+notice() {
+	echo "$@"
+	preinit_net_echo "$@"
+}
+
+user_config_menu() {
+	local user_config_options
+	local no_user_config
+	local user_config_wait_timeout
+
+	user_config_options=$(ls /etc/config-menu.d)
+	set -- $user_config_options
+	local options_limit=$#
+	no_user_config=$(uci -q get user.@config[0].no_user_config) ||
+		no_user_config=$pi_no_user_config
+	[ "$no_user_config" = "y" -o "$options_limit" -eq 0 ] && return
+
+	user_config_wait_timeout=$(uci -q get user.@config[0].user_config_wait_timeout) ||
+		user_config_wait_timeout=$pi_user_config_wait_timeout
+	[ -z "$user_config_wait_timeout" ] && user_config_wait_timeout=2
+	local buttons_hook=$(mktemp -d)
+	cp -pa /etc/rc.button/buttons "$buttons_hook"
+	mount --bind "$buttons_hook" /etc/rc.button/
+	/sbin/procd -h /etc/hotplug-config.json &
+	local procd_pid=$!
+
+	local undo="kill $procd_pid ; umount /etc/rc.button/ ; rm $buttons_hook/buttons && rmdir $buttons_hook"
+	trap "$undo" INT
+	trap "$undo" USR1
+
+	while true ; do
+		set_state config
+		notice "Please press the number or any button X times within [$user_config_wait_timeout] " \
+		       "seconds to switch the config:"
+		echo "0: Exit the menu and continue booting."
+		local NUMBER=0
+		for option in $user_config_options ; do
+			optionval=$(uci -q get user.@config[0].${option})
+			msg="$((++NUMBER)): $option=$optionval -> $((($optionval+1)%2))"
+			notice "$msg"
+		done
+		echo -n "Option: "
```

Comment:
Console text could be clearer... keypressed is fairly ok... but button operation... can be worded better... feedback on presses would be a bonus... as would changing the blink rate based on what was just enabled... but i know that's alot of work...


Existing values in uci activate the wait time / press loop again... but empty uci does not allow for setting multiple options via buttons... ( keyboard does ).... 

Well they are the main things... on a general level....  no rush to make any changes... just some feedback that may be of use....